pool.go

```diff
@@ -153,18 +158,33 @@ func (p *pool) Close() {
 	}
 }
 
-func (p *pool) fetchWith(ctx context.Context, c cid.Cid, with string) (blocks.Block, error) {
+func (p *pool) fetchWith(ctx context.Context, c cid.Cid, with string) (blk blocks.Block, err error) {
 	<-p.started
-	aff := with
-	if aff == "" {
-		aff = c.Hash().B58String()
+
+	left := p.config.MaxRetries
+	if left == 0 {
+		left = DefaultMaxRetries
 	}
-	p.lk.RLock()
-	member := p.c.LocateKey([]byte(aff))
-	p.lk.RUnlock()
-	root := member.String()
 
-	return p.doFetch(ctx, root, c)
+	for i := 0; i < left; i++ {
+		aff := with
+		if aff == "" {
+			aff = fmt.Sprintf("%d%s", i, c.Hash().B58String())
+		} else {
+			aff = fmt.Sprintf("%d%s", i, aff)
+		}
+		p.lk.RLock()
+		member := p.c.LocateKey([]byte(aff))
+		p.lk.RUnlock()
+		root := member.String()
```

Comment:
Can't we keep this outside the loop? Probably netgligable computational complexity, but still.

Reply:
we want to fail-over to a different backend when the first one fails - so we re-locate a different member by putting the different retry number into the key we locate